flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBaseConfig.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.cassandra;
+
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Configuration for {@link CassandraSinkBase}.
+ */
+public class CassandraSinkBaseConfig implements Serializable  {
+	// ------------------------ Default Configurations ------------------------
+
+	/**
+	 * The default maximum number of concurrent requests. By default, {@code Integer.MAX_VALUE}.
+	 */
+	public static final int DEFAULT_MAX_CONCURRENT_REQUESTS = Integer.MAX_VALUE;
+
+	/**
+	 * The default timeout duration when acquiring a permit to execute. By default, {@code Long.MAX_VALUE}.
+	 */
+	public static final long DEFAULT_MAX_CONCURRENT_REQUESTS_TIMEOUT = Long.MAX_VALUE;
+
+	/**
+	 * The default timeout unit when acquiring a permit to execute. By default, milliseconds.
+	 */
+	public static final TimeUnit DEFAULT_MAX_CONCURRENT_REQUESTS_TIMEOUT_UNIT = TimeUnit.MILLISECONDS;
+
+	// ------------------------- Configuration Fields -------------------------
+
+	/** Maximum number of concurrent requests allowed. */
+	private final int maxConcurrentRequests;
+
+	/** Timeout duration when acquiring a permit to execute. */
+	private final long maxConcurrentRequestsTimeout;
+
+	/** Timeout unit when acquiring a permit to execute. */
+	private final TimeUnit maxConcurrentRequestsTimeoutUnit;
+
+	public CassandraSinkBaseConfig(
+		int maxConcurrentRequests,
+		long maxConcurrentRequestsTimeout,
+		TimeUnit maxConcurrentRequestsTimeoutUnit) {
+		Preconditions.checkArgument(maxConcurrentRequests >= 0,
+			"Max concurrent requests is expected to be positive");
+		Preconditions.checkArgument(maxConcurrentRequestsTimeout >= 0,
+			"Max concurrent requests timeout is expected to be positive");
+		Preconditions.checkNotNull(maxConcurrentRequestsTimeoutUnit,
+			"Max concurrent requests timeout unit cannot be null");
+		this.maxConcurrentRequests = maxConcurrentRequests;
+		this.maxConcurrentRequestsTimeout = maxConcurrentRequestsTimeout;
+		this.maxConcurrentRequestsTimeoutUnit = maxConcurrentRequestsTimeoutUnit;
+	}
+
+	public int getMaxConcurrentRequests() {
+		return maxConcurrentRequests;
+	}
+
+	public long getMaxConcurrentRequestsTimeout() {
+		return maxConcurrentRequestsTimeout;
+	}
+
+	public TimeUnit getMaxConcurrentRequestsTimeoutUnit() {
+		return maxConcurrentRequestsTimeoutUnit;
+	}
+
+	@Override
+	public String toString() {
+		return "CassandraSinkBaseConfig{" +
+			"maxConcurrentRequests=" + maxConcurrentRequests +
+			", maxConcurrentRequestsTimeout=" + maxConcurrentRequestsTimeout +
+			", maxConcurrentRequestsTimeoutUnit=" + maxConcurrentRequestsTimeoutUnit +
+			'}';
+	}
+
+	public static Builder newBuilder() {
```

Comment:
either the constructor should be private or this method removed; there's no value in having 2 ways of creating the builder.

Reply:
This comment hasn't been addressed yet, the `Builder` constructor is still public.